validator-auction-rest-server/lib/EthEventsClient.js

```diff
@@ -19,31 +19,37 @@ export default class EthEventsClient {
     }
 
     async getAuctionSummary() {
-        const result = {
-            contractAddress: this._contractAddress
-        }
-        const [currentBlockTime, bids, auctionStart, whitelistedAddresses, deploymentParams] = await Promise.all([
-            this.getCurrentBlockTime(),
-            this.getBidEvents(),
-            this.getAuctionStartInSeconds(),
-            this.getWhitelistedAddresses(),
-            this.getAuctionDeploymentParameters()
-        ])
-
-        if(!deploymentParams || !auctionStart) {
-            result.auctionHasNotStarted = true
-            return result
+        const [currentBlockTime, allEvents] = await Promise.all([this.getCurrentBlockTime(), this.getAllEvents()])
+
+        const state = this.getAuctionState(allEvents)
+
+        if (state === 'Not Deployed') {
+            return {
+                auctionState: state
+            }
         }
 
-        result.whitelistedAddresses = whitelistedAddresses
-        result.bids = bids
-        result.takenSlotsCount = bids.length
-        result.freeSlotsCount = deploymentParams.numberOfParticipants - result.takenSlotsCount
-        result.remainingSeconds = EthEventsClient.calculateRemainingAuctionSeconds(auctionStart, currentBlockTime, deploymentParams.durationInDays)
-        result.currentPriceInWEI = EthEventsClient.getCurrentPriceAsBigNumber(auctionStart * 1000, currentBlockTime * 1000, deploymentParams.durationInDays, deploymentParams.startPrice).toString()
-        result.priceFunction = EthEventsClient.calculateAllSlotPrices(auctionStart, deploymentParams.durationInDays, deploymentParams.startPrice)
-        result.currentBlocktimeInMs = currentBlockTime * 1000
-        return result
+        const deploymentParams = this.getAuctionDeploymentParameters(allEvents)
+        const bids = this.getBids(allEvents)
+        const whitelistedAddresses = this.getWhitelistedAddresses(allEvents)
+        let auctionStart = this.getAuctionStartInSeconds(allEvents)
+        if (!auctionStart) {
+            auctionStart = currentBlockTime
+        }
+        const currentPriceInWEI = state === 'Finished' ? this.getClosingPrice(allEvents).toString() : EthEventsClient.getCurrentPriceAsBigNumber(auctionStart * 1000, currentBlockTime * 1000, deploymentParams.durationInDays, deploymentParams.startPrice).toString()
```

Comment:
What do we show for the other states? For example for Failed?

Reply:
When status is "Failed", the price is hidden on the UI.
When status is "Finished", the last slot price (given from the event) is rendered in the UI and the description changes to Closing Price.

Reply:
I would rather have the currentPrice then only calculated when it makes sense. That we dont show it means probably that the value does not make sense, so we should make that clear by also not calculating it here. 

So:
auction deployed: undefined
auction started: calculate price
auction finished: undefined
auction erroreed: undefined

for the closing price we can give it a new name.